compiler/rustc_codegen_llvm/src/common.rs

```diff
@@ -223,13 +222,13 @@ impl<'ll, 'tcx> ConstMethods<'tcx> for CodegenCx<'ll, 'tcx> {
         })
     }
 
+    fn zst_to_backend(&self, _llty: &'ll Type) -> &'ll Value {
+        self.const_undef(self.type_ix(0))
+    }
+
     fn scalar_to_backend(&self, cv: Scalar, layout: abi::Scalar, llty: &'ll Type) -> &'ll Value {
         let bitsize = if layout.is_bool() { 1 } else { layout.size(self).bits() };
         match cv {
-            Scalar::Int(ScalarInt::ZST) => {
-                assert_eq!(0, layout.size(self).bytes());
```

Comment:
So I'm looking at this (in the context of Rust-GPU updating the nightly version of Rust we use).

How could this ever pass? `abi::Scalar` just can't encode sizes other than 1, 2, 4, 8, 16:
https://github.com/rust-lang/rust/blob/3c7278846102bb829c9a789e91bc43f0ed612943/compiler/rustc_target/src/abi/mod.rs#L605-L611
https://github.com/rust-lang/rust/blob/3c7278846102bb829c9a789e91bc43f0ed612943/compiler/rustc_target/src/abi/mod.rs#L703-L715
https://github.com/rust-lang/rust/blob/3c7278846102bb829c9a789e91bc43f0ed612943/compiler/rustc_target/src/abi/mod.rs#L827-L844

Reply:
This is the const eval Scalar, not abi Scalar.

Reply:
`layout` is `abi::Scalar` though.

If you scroll up through this PR there is a comment of mine somewhere saying that this must have been dead code.

Reply:
`layout` is `abi::Scalar` though.

If you scroll up through this PR there is a comment of mine somewhere saying that this must have been dead code.